debugging/book-library/script.js

```diff
@@ -1,103 +1,103 @@
 
-//check the right input from forms and if its ok -> add the new book (object in array)
-//via Book function and start render function
-function submit() {
-  if (
-    title.value == null ||
-    title.value == "" ||
-    pages.value == null ||
-    pages.value == ""
-  ) {
-    alert("Please fill all fields!");
-    return false;
-  } else {
-    let book = new Book(title.value, title.value, pages.value, check.checked);
-    library.push(book);
-    render();
+  function submit() {
+    if (
+      title.value == null ||
+      title.value === "" ||
+      pages.value == null ||
+      pages.value === ""
+    ) {
+      alert("Please fill all fields!");
+      return false;
+    } else {
+      let book = new Book(
+        title.value,
+        author.value,
+        pages.value,
+        check.checked
+      );
+      myLibrary.push(book);
+      render();
+    }
   }
-}
 
-function Book(title, author, pages, check) {
-  this.title = title;
-  this.author = author;
-  this.pages = pages;
-  this.check = check;
-}
+  submitButton.addEventListener("click", submit);
 
-function render() {
-  let table = document.getElementById("display");
-  let rowsNumber = table.rows.length;
-  //delete old table
-  for (let n = rowsNumber - 1; n > 0; n-- {
-    table.deleteRow(n);
+  function Book(title, author, pages, check) {
+    this.title = title;
+    this.author = author;
+    this.pages = pages;
+    this.check = check;
   }
-  //insert updated row and cells
-  let length = myLibrary.length;
-  for (let i = 0; i < length; i++) {
-    let row = table.insertRow(1);
-    let cell1 = row.insertCell(0);
-    let cell2 = row.insertCell(1);
-    let cell3 = row.insertCell(2);
-    let cell4 = row.insertCell(3);
-    let cell5 = row.insertCell(4);
-    cell1.innerHTML = myLibrary[i].title;
-    cell2.innerHTML = myLibrary[i].author;
-    cell3.innerHTML = myLibrary[i].pages;
 
-    //add and wait for action for read/unread button
-    let changeBut = document.createElement("button");
-    changeBut.id = i;
-    changeBut.className = "btn btn-success";
-    cell4.appendChild(changeBut);
-    let readStatus = "";
-    if (myLibrary[i].check == false) {
-      readStatus = "Yes";
-    } else {
-      readStatus = "No";
+  function render() {
+    let table = document.getElementById("display");
+    // Clear the previous rows, excluding the header
+    const rowsNumber = table.rows.length;
+    for (let n = rowsNumber - 1; n > 0; n--) {
+      table.deleteRow(n);
     }
-    changeBut.innerHTML = readStatus;
 
-    changeBut.addEventListener("click", function () {
-      myLibrary[i].check = !myLibrary[i].check;
-      render();
-    });
+    // Insert updated rows and cells
+    myLibrary.forEach((book, i) => {
+      let row = table.insertRow();
+      let cell1 = row.insertCell(0);
+      let cell2 = row.insertCell(1);
+      let cell3 = row.insertCell(2);
+      let cell4 = row.insertCell(3);
+      let cell5 = row.insertCell(4);
```

Comment:
Note: These variables have been renamed in CYF's `main`.

Reply:
thanks, changed